pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -0,0 +1,172 @@
+package triggers
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+// TODO: Register the capiability in the NewDelegateFunction in chalink/core/services/workflows.delegate.go
+
+var mercuryInfo = capabilities.MustNewCapabilityInfo(
+	"mercury-trigger",
+	capabilities.CapabilityTypeTrigger,
+	"An example mercury trigger.",
+	"v1.0.0",
+)
+
+// This Trigger Service allows for the registration and deregistration of triggers. You can also send reports to the service.
+type MercuryTriggerService struct {
+	capabilities.CapabilityInfo
+	chans               map[workflowID]chan<- capabilities.CapabilityResponse
+	feedIdsForTriggerId map[string][]uint64
+	mu                  sync.Mutex
+	feedIdToWorkflowId  map[uint64]string
+}
+
+type MercuryReport struct {
+	feedId               uint64
+	fullreport           []byte
+	benchmarkPrice       int64
+	observationTimestamp uint32
+}
+
+type MercuryTriggerEvent struct {
+	triggerType string // "mercury"
+	id          string // "sha256 of payload.feedId + payload.timestamp"
+	timestamp   string // "current time"
+	payload     MercuryReport
+}
+
+var _ capabilities.TriggerCapability = (*MercuryTriggerService)(nil)
+
+func NewMercuryTriggerService() *MercuryTriggerService {
+	return &MercuryTriggerService{
+		CapabilityInfo:      mercuryInfo,
+		chans:               map[workflowID]chan<- capabilities.CapabilityResponse{},
+		feedIdsForTriggerId: make(map[string][]uint64),
+		feedIdToWorkflowId:  make(map[uint64]string),
+	}
+}
+
+// maybe use mercury-pipline reports.go report struct instead of MercuryReport struct, and then we can use the pipeline report generators
+func (o *MercuryTriggerService) ProcessReport(report MercuryReport) error {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+
+	currentTime := time.Now()
+	unixTimestampMillis := currentTime.UnixNano() / int64(time.Millisecond)
+	feedId := report.feedId
+
+	// Create a new CapabilityResponse with the MercuryTriggerEvent
+	triggerEvent := MercuryTriggerEvent{
+		triggerType: "mercury",
+		id:          sha256Hash(strconv.FormatUint(feedId, 10) + strconv.FormatUint(uint64(report.observationTimestamp), 10)),
```

Comment:
I think it'd be good to extract this into a `generateTriggerEventId` method. I'd imagine we'd want this to eventually be part of any trigger service, since its a requirement for trigger events to have a unique ID. This will also make testing this function easier